packages/protocol-parser/src/abiTypesToSchemaData.ts

```diff
@@ -0,0 +1,21 @@
+import { Hex } from "viem";
+import { DynamicAbiType } from "./dynamicAbiTypes";
+import { schemaAbiTypes } from "./schemaAbiTypes";
+import { StaticAbiType, staticAbiTypeToByteLength } from "./staticAbiTypes";
+
+export function abiTypesToSchemaData(staticFields: StaticAbiType[], dynamicFields: DynamicAbiType[] = []): Hex {
```

Comment:
thoughts on calling this `EncodedSchema` instead of `SchemaData`?

Reply:
I originally had this as `rawSchema` but using `data` seemed to align better with everything else (each helper in this package that accepts hex arg named "data", viem calls hex "data" in many helpers)

could also see this being `abiTypesToSchemaHex ` instead and using `hex` everywhere

since each helper is pretty well contained, having consistent naming made it a little easier to move logic around into different places, otherwise we'd have hex with a bunch of different names: `encodedSchema`, `encodedTableSchema`, `encodedFieldValue`, `encodedKeyTuple`, etc.

Reply:
re: naming, I was also stuck on whether things like `decodeDynamicField` should be named differently, since it's not a pure "hex to something" but rather "abi type + hex to its JS primitive type"